src/rhsm/m2cryptohttp.py

```diff
@@ -102,7 +103,11 @@ def __init__(self, host, *args, **kwargs):
 
     def _start_ssl(self):
         self.sock = SSL.Connection(self.ssl_ctx, self.sock)
-        self.sock.settimeout(self.rhsm_timeout)
+        try:
+            self.sock.settimeout(self.rhsm_timeout)
+        except AttributeError:
+            sock.set_socket_write_timeout(timeout(self.rhsm_timeout))
+            sock.set_socket_read_timeout(timeout(self.rhsm_timeout))
```

Comment:
There should be:

```python
self.sock.set_socket_write_timeout(timeout(self.rhsm_timeout))
self.sock.set_socket_read_timeout(timeout(self.rhsm_timeout))
```

Reply:
ea9a1a7e6..a5a94100f: address review feedback

@jirihnidek nice catch! thanks!